### DISCUSSION TITLE
Ability to hava reference by name (in this case Synthetic Location) without hardcoding the Location ID

### DISCUSSION THREAD

**Discussion by nsaputro-shell** [Dynatrace/dynatrace-configuration-as-code#1304] [General]
Hi,

I tried creating template for Synthetic monitor and it requires location id.

Is there a way to have reference by name instead? I read the documentation  but I could not find a solution for it.

With Dynatrace Terraform provider it could be easily be done using `dynatrace_synthetic_location `

```
data "dynatrace_synthetic_location" "Test" {
  cloud_platform = "AMAZON_EC2"
  type           = "PUBLIC"
  name           = "Sydney"
}

resource "dynatrace_http_monitor" "monitor" {
  name = "test-monitor" 
  enabled = true 
  frequency = 60 
  locations = ["${data.dynatrace_synthetic_location.Test.id}"]
```

Do we have similar thing with Monaco ?

**Comment by UnseenWizzard** [comment]
Hi @nsaputro-shell , 

no, this is sadly not possible in Monaco.
We have nothing matching Terraform's Data Sources pulling in data from the environment. 

You'll have to look up the ID either by checking a downloaded synthetic provider or using the API.

**Reply by nsaputro-shell** [reply]
@UnseenWizzard Is there any plan to support it as this is very useful so we do need need to perform API call on each environment to get resource id? Maybe by introducing a new type of reference parameter that retrieves ID by providing some selectors/criteria.

**Reply by UnseenWizzard** [reply]
@nsaputro-shell there is currently no plan to introduce this for synthetics specifically.
There is https://github.com/Dynatrace/dynatrace-configuration-as-code/issues/964 for allowing more generalized entity selectors - I think this could cover synthetic locations - or should be extended to cover them if that API does not (FYI @Laubi). 
At this point, I can't give you any ETA for this feature though.

**Reply by nsaputro-shell** [reply]
Thanks @UnseenWizzard for the explanation.

Do you have suggestion which tool to use, Terraform or Monaco? We are still in the early phase so switching between them is not a big deal.

Dynatrace Terraform provider has some nice features that is not available in Monaco. I see also Terraform uses some newer API and Monaco still uses the deprecated one for example credentials vault (v2 vs v1).

Will there be a shifting that Terraform will be officially supported and Monaco will be deprecated? or Monaco will also be updated to use newer APIs ?

**Reply by UnseenWizzard** [reply]
Hi @nsaputro-shell, yes, Terraform will be officially supported in the near future. 
As it covers what you need, I recommend using Terraform rather than waiting for future Monaco features.

In general, though, we aim to keep API support roughly on the same level for both - and upgrade to newer API versions when needed. 
Is there a specific feature of the credentials vault v2 API you're missing in Monaco?

**Reply by nsaputro-shell** [reply]
@UnseenWizzard No specific feature/issue for credentials vault. I just found the API reference difference when comparing Terraform and Monaco.
